nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,459 @@
+            .displayName("Retry timeout")
+            .description("Controls the amount of time, in milliseconds, before a timeout occurs when retrying the operation.")
+            .required(true)
+            .defaultValue("60000")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final AllowableValue BREAK_UP_YES = new AllowableValue(
+        "breakup-yes",
+        "Yes",
+        "Break up results."
+    );
+    public static final AllowableValue BREAK_UP_HITS_NO = new AllowableValue(
+        "breakup-no",
+        "No",
+        "Don't break up results."
+    );
+
+    public static final PropertyDescriptor BREAK_UP_HITS = new PropertyDescriptor.Builder()
+            .name("el5-break-up-hits")
+            .displayName("Break up search results")
+            .description("Break up search results into one flowfile per result.")
+            .allowableValues(BREAK_UP_HITS_NO, BREAK_UP_YES)
+            .defaultValue(BREAK_UP_HITS_NO.getValue())
+            .required(true)
+            .expressionLanguageSupported(false)
+            .build();
+    public static final PropertyDescriptor BREAK_UP_AGGREGATIONS = new PropertyDescriptor.Builder()
+            .name("el5-break-up-aggregations")
+            .displayName("Break up aggregation results")
+            .description("Break up aggregation results into one flowfile per result.")
+            .allowableValues(BREAK_UP_HITS_NO, BREAK_UP_YES)
+            .defaultValue(BREAK_UP_HITS_NO.getValue())
+            .required(true)
+            .expressionLanguageSupported(false)
+            .build();
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    static {
+        final Set<Relationship> _rels = new HashSet<>();
+        _rels.add(REL_SUCCESS);
+        _rels.add(REL_FAILURE);
+        _rels.add(REL_HITS);
+        _rels.add(REL_AGGREGATIONS);
+        relationships = Collections.unmodifiableSet(_rels);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(PROP_SSL_CONTEXT_SERVICE);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(SOCKET_TIMEOUT);
+        descriptors.add(RETRY_TIMEOUT);
+        descriptors.add(USERNAME);
+        descriptors.add(PASSWORD);
+        descriptors.add(HTTP_HOSTS);
+        descriptors.add(QUERY);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(BREAK_UP_HITS);
+        descriptors.add(BREAK_UP_AGGREGATIONS);
+
+        propertyDescriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+
+    @OnUnscheduled
+    public void onStartup() {
```

Comment:
Just a style note, but this might be a misnamed/confusing method for an @OnUnscheduled annotation